scripts/py2md.py

```diff
@@ -0,0 +1,226 @@
+                    if name.startswith("allennlp."):
+                        path = name.split(".")
+                        if ty == "mod":
+                            href = "/api/" + "/".join(path[1:])
+                        else:
+                            href = "/api/" + "/".join(path[1:-1]) + "#" + path[-1].lower()
+                        cross_ref = f"[`{path[-1]}`]({href})"
+                    elif "." not in name:
+                        cross_ref = f"[`{name}`](#{name.lower()})"
+                    else:
+                        cross_ref = f"`{name}`"
+                    line = line.replace(match, cross_ref)
+            lines.append(line)
+        node.docstring = "\n".join(lines)
+
+    def _preprocess_line(self, line, current_section):
+        match = re.match(r"#+ (.*)$", line)
+        if match:
+            current_section = match.group(1).strip().lower()
+            line = re.sub(r"#+ (.*)$", r"__\1__\n", line)
+        else:
+            if line and not line.startswith(" "):
+                if (
+                    current_section
+                    in ("arguments", "parameters", "attributes", "members", "returns")
+                    and ":" in line
+                ):
+                    ident, ty = line.split(":", 1)
+                    if ty:
+                        line = f"- __{ident}__ : {ty}<br>"
+                    else:
+                        line = f"- __{ident}__ :<br>"
+                elif current_section in ("returns", "raises"):
+                    line = f"- {line} <br>"
+
+        return line, current_section
+
+
+@implements(Processor)
+class AllenNlpFilterProcessor(Struct):
+    """
+    Used to filter out nodes that we don't want to document.
+    """
+
+    SPECIAL_MEMBERS = ("__path__", "__annotations__", "__name__", "__all__", "__init__")
+
+    def process(self, graph, _resolver):
+        graph.visit(self._process_node)
+
+    def _process_node(self, node):
+        def _check(node):
+            if node.parent and node.parent.name.startswith("_"):
+                return False
+            if node.name.startswith("_") and not node.name.endswith("_"):
+                return False
+            if node.name in self.SPECIAL_MEMBERS:
+                return False
+            if node.name == "logger" and isinstance(node.parent, Module):
+                return False
+            return True
+
+        if not _check(node):
+            node.visible = False
+
+
+@implements(Renderer)
+class AllenNlpRenderer(MarkdownRenderer):
+    def _format_function_signature(
+        self, func: Function, override_name: str = None, add_method_bar: bool = True
+    ) -> str:
+        parts = []
+        for dec in func.decorators:
+            parts.append("@{}{}\n".format(dec.name, dec.args or ""))
+        if self.signature_python_help_style and not func.is_method():
+            parts.append("{} = ".format(func.path()))
+        if func.is_async:
+            parts.append("async ")
```

Comment:
Do we have `async` functions anywhere? How did you decide to add this?

Reply:
This comes from here https://github.com/NiklasRosenstein/pydoc-markdown/blob/develop/src/pydoc_markdown/contrib/renderers/markdown.py#L36. Even though we don't have async functions anywhere at the moment I figured it would be good to keep, especially if we decide to spin this thing off into a more general tool in the future.